### ISSUE TITLE
Remove old data model, parsers and writers in new refactor branch

### ISSUE THREAD

**Issue by nicoweidner** [spdx/tools-python#358]
With the 0.7.0 release done, we can remove the old (and to be replaced) structures on the refactoring branch and get it merged into `main`, to avoid a "big bang" branch building up in parallel to `main`

**Comment by nicoweidner** [comment]
Fixed via https://github.com/spdx/tools-python/pull/361
